bigquery/biglake/bigquery_create_partitioned_biglake_table/main.tf

```diff
@@ -0,0 +1,119 @@
+* https://cloud.google.com/bigquery/docs/create-cloud-storage-table-biglake
+* and
+* https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/bigquery_table
+*/
+
+# [START bigquery_create_partitioned_biglake_table]
+# This creates a bucket in the US region named "my-bucket" with a pseudorandom suffix.
+resource "random_id" "bucket_name_suffix" {
+  byte_length = 8
+}
+resource "google_storage_bucket" "default" {
+  name                        = "my-bucket-${random_id.bucket_name_suffix.hex}"
+  location                    = "US"
+  force_destroy               = true
+  uniform_bucket_level_access = true
+}
+
+resource "google_storage_bucket_object" "default" {
+  # This creates a fake message to create partition locations on the table.
+  # Otherwise, the table deployment fails.
+  name    = "publish/dt=2000-01-01/hr=00/min=00/fake_message.json"
+  content = "{\"column1\": \"XXX\"}"
+  bucket  = google_storage_bucket.default.name
+}
+
+# This queries the provider for project information.
+data "google_project" "project" {}
+
+# This creates a connection in the US region named "my-connection".
+# This connection is used to access the bucket.
+resource "google_bigquery_connection" "default" {
+  connection_id = "my-connection"
+  location      = "US"
+  cloud_resource {}
+}
+
+# This grants the previous connection IAM role access to the bucket.
+resource "google_project_iam_member" "default" {
+  role    = "roles/storage.objectViewer"
+  project = data.google_project.project.id
+  member  = "serviceAccount:${google_bigquery_connection.default.cloud_resource[0].service_account_id}"
+}
+
+# This makes the script wait for seven minutes before proceeding.
+# This lets IAM permissions propagate.
+resource "time_sleep" "wait_7_min" {
+  depends_on      = [google_project_iam_member.default]
+  create_duration = "7m"
+}
+
+# This defines a Google BigQuery dataset with
+# default expiration times for partitions and tables, a
+# description, a location, and a maximum time travel.
+resource "google_bigquery_dataset" "default" {
+  dataset_id                      = "my_dataset"
+  default_partition_expiration_ms = 2592000000  # 30 days
+  default_table_expiration_ms     = 31536000000 # 365 days
+  description                     = "My dataset description"
+  location                        = "US"
+  max_time_travel_hours           = 96 # 4 days
+
+  # This defines a map of labels for the bucket resource,
+  # including the labels "billing_group" and "pii".
+  labels = {
+    billing_group = "accounting",
+    pii           = "sensitive"
+  }
+}
+
+# This creates a BigQuery table named "my_table" in the dataset "default".
+# The table has a single column named "column1", which is of type STRING
+# and is nullable.
+resource "google_bigquery_table" "default" {
+  dataset_id = google_bigquery_dataset.default.dataset_id
+  table_id   = "my_table"
+  schema     = jsonencode([{ "name" : "column1", "type" : "STRING", "mode" : "NULLABLE" }])
```

Comment:
Could you double check that this is working as intended? All the examples I saw had the TF syntax of using "=" instead of ":" in the input

Reply:
A. Partitioned Table. I ran the up to date script in cloud shell. Resulting Schema in Pantheon
![image](https://github.com/terraform-google-modules/terraform-docs-samples/assets/6531363/36fd6362-227e-4244-a8ff-3b8aa3f992de)

B. Unpartitioned Table. I ran the up to date script in cloud shell. Resulting Schema in Pantheon
![image](https://github.com/terraform-google-modules/terraform-docs-samples/assets/6531363/9cef5872-34cf-4635-986f-dc970c2a381e)